### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by heaven** [mhenrixon/sidekiq-unique-jobs#631] [Q&A] [answered]
Hi, I'm having issues with Sidekiq's test mode. It seems the unique jobs extension still reaches Redis and creates locks there.

I think that's not right and one may think they don't touch Redis in the test environment while in fact, they do. I am currently facing a problem with `MyWorker.jobs.clear`. I'm clearing the jobs but can't enqueue a new job until I do `Sidekiq.redis(&:flushdb)`.

So in my use case, we create a record and it pushes a job. Then I need to delete the job, make a few adjustments to the record, and make sure a new job is pushed once the record is updated.

**Comment by mhenrixon** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/106795/128556284-a3b052eb-37e8-4a10-a523-76f007eed372.jpeg)

**Comment by heaven** [comment]
Thank you, I saw this but the link doesn't open so I was hoping there are other solutions.
